monix-catnap-micrometer/src/main/scala/com/avast/sst/monix/catnap/micrometer/MicrometerCircuitBreakerMetricsModule.scala

```diff
@@ -0,0 +1,44 @@
+package com.avast.sst.monix.catnap.micrometer
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import cats.effect.Sync
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State.{Closed, HalfOpen, Open}
+import io.micrometer.core.instrument.MeterRegistry
+
+object MicrometerCircuitBreakerMetricsModule {
+
+  /** Makes [[com.avast.sst.monix.catnap.CircuitBreakerMetrics]] from [[io.micrometer.core.instrument.MeterRegistry]]. */
+  def make[F[_]: Sync](name: String, meterRegistry: MeterRegistry): F[CircuitBreakerMetrics[F]] = {
+    Sync[F].delay(new MicrometerCircuitBreakerMetrics(name, meterRegistry))
+  }
+
+  private class MicrometerCircuitBreakerMetrics[F[_]: Sync](name: String, meterRegistry: MeterRegistry) extends CircuitBreakerMetrics[F] {
+
+    private val F = Sync[F]
+
+    private[this] val CircuitOpened = -1
+    private[this] val CircuitHalfOpened = 0
+    private[this] val CircuitClosed = 1
+
+    private val accepted = meterRegistry.counter(s"circuit-breaker.$name.accepted")
+    private val rejected = meterRegistry.counter(s"circuit-breaker.$name.rejected")
+    private val circuitState = meterRegistry.gauge[AtomicInteger](s"circuit-breaker.$name.state", new AtomicInteger(CircuitClosed))
```

Comment:
what do you think about using the `IORef`?

Reply:
This has already been discussed. Gauge requries `Number` so it's not worth the complication to introduce `IORef`.